engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/WarningBenchmarks.java

```diff
@@ -28,27 +32,50 @@
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
```

Comment:
Is the warmup *sufficient*? I have a minor suggestion. You don't have to do this as I plan to do that in the future for all the benchmarks, but it would help with the stability of these benchmarks.

Modify the context to include an `option` for `-D.engine.TraceCompilationDetails=true` and check whether the output contains any "[engine]" logs, i.e., whether there are any Graal/Truffle compilations going on in the measurement phase. I guess it is enough to redirect the output to the stdout and just look when did the warmup phase ended and whether there are some logs from Graal. If you struggle with that too much, don't do that.

Reply:
An inspiration how to do that easily is here: https://github.com/enso-org/enso/issues/6271#issuecomment-1507924937

Reply:
Sorry, I missed that comment.